pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -18,7 +18,49 @@ package v1beta1
 
 import (
 	"context"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/knative/serving/pkg/apis/config"
 )
 
+// SetDefaults implements apis.Defaultable
 func (r *Revision) SetDefaults(ctx context.Context) {
+	r.Spec.SetDefaults(ctx)
+}
+
+// SetDefaults implements apis.Defaultable
+func (rts *RevisionTemplateSpec) SetDefaults(ctx context.Context) {
+	rts.Spec.SetDefaults(ctx)
+}
+
+// SetDefaults implements apis.Defaultable
+func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
+	cfg := config.FromContextOrDefaults(ctx)
+
+	// Default TimeoutSeconds based on our configmap
+	if rs.TimeoutSeconds == nil {
+		ts := cfg.Defaults.RevisionTimeoutSeconds
+		rs.TimeoutSeconds = &ts
+	}
+
+	var container corev1.Container
+	if len(rs.PodSpec.Containers) == 1 {
```

Comment:
Why the logic here? Shouldn't we expect this to be validated to be only 1? If we can't expect this to be validated why would we not error?

Reply:
I believe we run `SetDefaults()` before `Validate()`, so this needs to be a bit resilient.